x-pack/plugins/observability/server/services/slo/slo_repository.ts

```diff
@@ -36,7 +41,14 @@ export class KibanaSavedObjectsSLORepository implements SLORepository {
   }
 
   async deleteById(id: string): Promise<void> {
-    await this.soClient.delete(SO_SLO_TYPE, id);
+    try {
+      await this.soClient.delete(SO_SLO_TYPE, id);
+    } catch (err) {
+      if (SavedObjectsErrorHelpers.isNotFoundError(err)) {
+        return;
+      }
```

Comment:
💬 Considering a not found as a success operation but we might want to handle the error differently later.